pupil_src/shared_modules/video_capture/file_backend.py

```diff
@@ -165,119 +206,10 @@ def run_func(self, *args, **kwargs):
 
         return decorator
 
-    def _get_containers_pattern_lst(self, source_path):
-        '''
-        Return an order list sort by file name pattern
-        '''
-        # [world.mjpeg, world_001.mjpeg, world_002.mjpeg]
-        pattern, suffix = os.path.splitext(source_path)
-        return sorted(glob.glob(pattern + '*' + suffix))
-
-    def _get_timestamps_pattern_lst(self, source_path):
-        # [world_timestamps.npy, world_001_timestamps.npy,
-        # world_002_timestamps.npy]
-        pattern, _ = os.path.splitext(source_path)
-        suffix = "_timestamps.npy"
-        tmp = sorted(glob.glob(pattern + '*' + suffix))
-        # After sorted, tmp will return [world_001_timestamps.npy,
-        # world_002_timestamps.npy, world_timestamps.npy]
-        # So we have to reorder it
-        return tmp[-1:] + tmp[:-1]
-
-    def _get_conatiners_path(self, source_path):
-        return self._get_containers_pattern_lst(source_path)
-
-    def _get_containers(self, index):
-        '''
-        Get container by index, get the next one if the prev one is broken.
-        '''
-        try:
-            container = av.open(self.containers_path[index])
-        except av.AVError:
-            logger.info(
-                "Video at {0} is broken".format(self.containers_path[index]))
-            # return the next container
-            self.current_container_index += 1
-            return self._get_containers(self.current_container_index)
-        except IndexError:
-            raise NoMoreVideoError("No more container found")
-        else:
-            return container
-
-    def _set_timestamps(self, source_path):
-        '''
-        concatenate timestamp files (actual timestamps)
-        calculate np.diff and find indeces that represent gaps bigger than X
-        '''
-        self.timestamps_lst = self._get_timestamps_pattern_lst(
-            source_path)
-        avg_rate = self.video_stream.average_rate
-        if avg_rate is None:
-            avg_rate = Fraction(0, 1)
-
-        if float(avg_rate) % 1 != 0.0:
-            logger.error(
-                "Videofile pts are not evenly spaced, pts to" +
-                "index conversion may fail and be inconsitent."
-            )
-        self.timestamps = np.array([])
-        try:
-            for ts_filename in self.timestamps_lst:
-                timestamp = np.load(ts_filename)
-                self.frame_count = np.append(
-                    self.frame_count, len(timestamp))
-                self.timestamps = np.append(
-                    self.timestamps, timestamp)
-        except IOError:
```

Comment:
1. We assume the timestamps file to be there. If it is not, the recording is not valid.
2. `isinstance(np.float64(X), float) -> True`. Since `lookup.timestamp` is of type `np.float64`, the instance test became obsolete.